src/System Application/App/Agent/Setup/AgentAccessControls/SelectAgentAccessControl.Page.al

```diff
@@ -0,0 +1,70 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Agents;
+
+page 4321 "Select Agent Access Control"
+{
+    PageType = StandardDialog;
+    ApplicationArea = All;
+    SourceTable = "Agent Access Control";
+    SourceTableTemporary = true;
+    Caption = 'Select users to manage tasks and configure the agent';
+    Extensible = false;
+    DataCaptionExpression = '';
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    layout
+    {
+        area(Content)
+        {
+            part(AccessControlPart; "Select Agent Access Ctrl Part")
+            {
+                Caption = 'Agent Access Control';
+                ApplicationArea = All;
+            }
+        }
+    }
+
+    trigger OnOpenPage()
+    begin
+        BackupAgentAccessControl();
+        CurrPage.AccessControlPart.Page.Initialize(AgentUserSecurityID, Rec);
+    end;
+
+    trigger OnQueryClosePage(CloseAction: Action): Boolean
+    begin
+        if CloseAction = CloseAction::LookupCancel then
+            RestoreAgentAccessControl();
+
+        exit(true);
+    end;
+
+    internal procedure Initialize(NewAgentUserSecurityID: Guid; var TempAgentAccessControl: Record "Agent Access Control" temporary)
+    begin
+        AgentUserSecurityID := NewAgentUserSecurityID;
+        Rec.Copy(TempAgentAccessControl, true);
+    end;
+
+    internal procedure GetTempAgentAccessControl(var TempAgentAccessControl: Record "Agent Access Control" temporary)
+    begin
+        TempAgentAccessControl.Copy(Rec, true);
+    end;
+
+    local procedure BackupAgentAccessControl()
+    begin
+        TempBackupAgentAccessControl.Copy(Rec, true);
```

Comment:
Same comment here, check if it is a deep copy or not

Reply:
Actually, you were right, I reverted now!